src/bundle.c

```diff
@@ -21,6 +21,9 @@ static gboolean mksquashfs(const gchar *bundlename, const gchar *contentdir, GEr
 	g_autoptr(GSubprocess) sproc = NULL;
 	GError *ierror = NULL;
 	gboolean res = FALSE;
+	gint mksquashfs_argcp;
+	gchar **mksquashfs_argvp = NULL;
```

Comment:
You could move this into the if body as it is only used there.

At the moment, this also leaks as it is not freed anywhere. The simplest would be to use glibs autofree helper here and replace the above by

```suggestion
	g_auto(GStrv) mksquashfs_argvp = NULL;
```

(see https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#GStrv)

Reply:
Hrm. Did this change but did notice this text
> This macro can be used to avoid having to do explicit cleanups of local variables when exiting functions. It often vastly simplifies handling of error conditions, removing the need for various tricks such as goto out or repeating of cleanup code. It is also helpful for non-error cases.

Does this mean the g_propagate_prefixed_error inside the if *shouldn't* use goto out?
Things appear to work as is, but I'm unclear on why the `goto out` is required to start with *shrug

Reply:
It means you don't have to use `goto out` in many cases. The goto is only used for the central `r_context_end_step("mksquashfs", res);` here.  Potentially this could also be repeated in each goto and the goto then replaced by an instant return statement.